x/mongo/driver/operation.go

```diff
@@ -273,6 +273,9 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 	var original error
 	var retries int
 	retryable := op.retryable(desc.Server)
+	if op.Client != nil {
```

Comment:
Can you add a comment about why we're doing it this way? My understanding is that we want to "reset" the session state for retryable writes because it will be set to `true` in the conditional statement below if this specific operation is retryable.